test/recipes/81-test_cmp_cli.t

```diff
@@ -0,0 +1,294 @@
+                start_mock_server("") || die "Cannot start CMP mock server";
+            }
+        }
+        foreach my $aspect (@all_aspects) {
+            $aspect = chop_dblquot($aspect);
+            next if $server_name eq "Mock" && $aspect eq "certstatus";
+            load_config($server_name, $aspect); # update with any aspect-specific settings
+            indir $server_name => sub {
+                my $tests = load_tests($server_name, $aspect);
+                test_cmp_cli_aspect($server_name, $aspect, $tests);
+            };
+        };
+        stop_mock_server() if $launch_mock;
+    };
+};
+
+close($faillog) if $faillog;
+
+sub load_tests {
+    my $server_name = shift;
+    my $aspect = shift;
+    my $test_config = $ENV{CMP_CONFIG} // "test_$server_name.cnf";
+    my $file = data_file("test_$aspect.csv");
+    my @result;
+
+    open(my $data, '<', $file) || die "Cannot open $file for reading: $!";
+  LOOP:
+    while (my $line = <$data>) {
+        chomp $line;
+        $line =~ s{\r\n}{\n}g; # adjust line endings
+        $line =~ s{_CA_DN}{$ca_dn}g;
+        $line =~ s{_SERVER_DN}{$server_dn}g;
+        $line =~ s{_SERVER_CN}{$server_cn}g;
+        $line =~ s{_SERVER_IP}{$server_ip}g;
+        $line =~ s{_SERVER_PORT}{$server_port}g;
+        $line =~ s{_SERVER_TLS}{$server_tls}g;
+        $line =~ s{_SERVER_PATH}{$server_path}g;
+        $line =~ s{_SERVER_CERT}{$server_cert}g;
+        $line =~ s{_KUR_PORT}{$kur_port}g;
+        $line =~ s{_PBM_PORT}{$pbm_port}g;
+        $line =~ s{_PBM_REF}{$pbm_ref}g;
+        $line =~ s{_PBM_SECRET}{$pbm_secret}g;
+        my $noproxy = $line =~ m/,\s*-no_proxy\s*,(.*?)(,|$)/ ? $1 : $no_proxy;
+        next LOOP if $no_proxy && ($noproxy =~ $server_cn || $noproxy =~ $server_ip)
+            && $line =~ m/,\s*-proxy\s*,/;
+        next LOOP if $server_tls == 0 && $line =~ m/,\s*-tls_used\s*,/;
+        $line =~ s{-section,,}{-section,,-proxy,$proxy,} unless $line =~ m/,\s*-proxy\s*,/;
+        $line =~ s{-section,,}{-config,../$test_config,-section,$server_name $aspect,};
+        my @fields = grep /\S/, split ",", $line;
+        s/^<EMPTY>$// for (@fields); # used for proxy=""
+        s/^\s+// for (@fields); # remove leading whitespace from elements
+        s/\s+$// for (@fields); # remove trailing whitespace from elements
+        s/^\"(\".*?\")\"$/$1/ for (@fields); # remove escaping from quotation marks from elements
+        my $expected_exit = $fields[$column];
+        my $description = 1;
+        my $title = $fields[$description];
+        next LOOP if (!defined($expected_exit)
+                      || ($expected_exit ne 0 && $expected_exit ne 1));
+        @fields = grep {$_ ne 'BLANK'} @fields[$description + 1 .. @fields - 1];
+        push @result, [$title, \@fields, $expected_exit];
+    }
+    close($data);
+    return \@result;
+}
+
+sub mock_server_pid {
+    return `lsof -iTCP:$localport -sTCP:LISTEN | tail -n 1 | awk '{ print \$2 }'`;
```

Comment:
As per comment below this is very UNIX specific

Reply:
Yes. At least I managed to make it run well also with Mac OS.
And the script skips on Win32.
To be on the safe side, I've just made it skip also on VMS.